src/ThemedWrappers.h

```diff
@@ -0,0 +1,154 @@
+   ThemedWindowWrapper(Args&&... args) : WindowBase( std::forward<Args>(args)... )
+   {
+      mThemeChangeSubscription =
+         theTheme.Subscribe(*this, &ThemedWindowWrapper::OnThemeChange);
+   }
+
+   void SetBackgroundColorIndex(int index)
+   {
+      mBackgroundColorIndex = index;
+      if(index != -1)
+         WindowBase::SetBackgroundColour(theTheme.Colour(mBackgroundColorIndex));
+   }
+
+   void SetForegroundColorIndex(int index)
+   {
+      mForegroundColorIndex = index;
+      if(index != -1)
+         WindowBase::SetForegroundColour(theTheme.Colour(mForegroundColorIndex));
+   }
+
+   void SetTranslatableLabel(TranslatableString label)
+   {
+      mTranslatableLabel = std::move(label);
+      WindowBase::SetLabel(mTranslatableLabel->Translation());
+   }
+
+   void UpdatePrefs() override
+   {
+      if(mTranslatableLabel)
+         WindowBase::SetLabel(mTranslatableLabel->Translation());
+   }
+
+protected:
+   void OnThemeChange(ThemeChangeMessage message)
+   {
+      if (message.appearance)
+      {
+         if(mBackgroundColorIndex != -1)
+            WindowBase::SetBackgroundColour(theTheme.Colour(mBackgroundColorIndex));
+         if(mForegroundColorIndex != -1)
+            WindowBase::SetForegroundColour(theTheme.Colour(mForegroundColorIndex));
+         WindowBase::Refresh();
+      }
+   }
+};
+
+template<class ButtonBase>
+class ThemedButtonWrapper final :
+   public ButtonBase,
+   public PrefsListener
+{
+   Observer::Subscription mThemeChangeSubscription;
+   int mLabelBitmapIndex { -1 };
+   int mPressedBitmapIndex { -1 };
+   int mCurrentBitmapIndex { -1 };
+   int mBackgroundColorIndex { -1 };
+   std::optional<TranslatableString> mTranslatableLabel;
+
+public:
+   template <typename... Args>
+   ThemedButtonWrapper(Args&&... args) : ButtonBase( std::forward<Args>(args)... )
+   {
+      mThemeChangeSubscription =
+         theTheme.Subscribe(*this, &ThemedButtonWrapper::OnThemeChange);
+   }
+
+   void SetBitmapLabelIndex(int index)
+   {
+      mLabelBitmapIndex = index;
+      ButtonBase::SetBitmapLabel(theTheme.Bitmap(mLabelBitmapIndex));
+   }
+   void SetBitmapPressedIndex(int index)
+   {
+      mPressedBitmapIndex = index;
+      ButtonBase::SetBitmapPressed(theTheme.Bitmap(mPressedBitmapIndex));
+   }
+
+   void SetBitmapCurrentIndex(int index)
+   {
+      mCurrentBitmapIndex = index;
+      ButtonBase::SetBitmapCurrent(theTheme.Bitmap(mCurrentBitmapIndex));
+   }
+
+   void SetBackgroundColorIndex(int index)
+   {
+      mBackgroundColorIndex = index;
+      if(index != -1)
+         ButtonBase::SetBackgroundColour(theTheme.Colour(mBackgroundColorIndex));
+   }
+
+   void SetTranslatableLabel(TranslatableString label)
+   {
+      mTranslatableLabel = std::move(label);
+      ButtonBase::SetLabel(mTranslatableLabel->Translation());
+   }
+
+   void UpdatePrefs() override
+   {
+      if(mTranslatableLabel)
+         ButtonBase::SetLabel(mTranslatableLabel->Translation());
```

Comment:
Ditto.

Preference changes might be written to send messages like theme changes, but it was implemented this way with virtual functions before Observer existed.  That rewrite would eliminate the problem of missed call throughs but would be the bigger work now.